automated install/basic-install.sh

```diff
@@ -190,24 +190,29 @@ os_check() {
 
         IFS=" " read -r -a supportedOS < <(dig +short -t txt ${remote_os_domain} @ns1.pi-hole.net | tr -d '"')
 
-        for i in "${supportedOS[@]}"
-        do
-            os_part=$(echo "$i" | cut -d '=' -f1)
-            versions_part=$(echo "$i" | cut -d '=' -f2-)
-
-            if [[ "${detected_os}" =~ ${os_part} ]]; then
-            valid_os=true
-            IFS="," read -r -a supportedVer <<<"${versions_part}"
-            for x in "${supportedVer[@]}"
-            do
-                if [[ "${detected_version}" =~ $x ]];then
-                valid_version=true
+        if [ -z "$supportedOS" ]; then
```

Comment:
Might be best here to do a count of the items in the array, rather than attempting to expand the first element

```suggestion
        if [  ${#supportedOS[@]} -eq 0 ]; then
```

I think the above should do it, but may want to double check!


Relevant detail from stickler:
https://github.com/koalaman/shellcheck/wiki/SC2128

Reply:
I'll go with what you prefer. Pushed the change.